libhb/decavcodec.c

```diff
@@ -1880,8 +1882,8 @@ static int decavcodecvInit( hb_work_object_t * w, hb_job_t * job )
         pv->context->opaque = job;
         av_buffer_replace(&pv->context->hw_device_ctx, w->hw_device_ctx);
 
-        if (job == NULL ||
-            (job->hw_pix_fmt == AV_PIX_FMT_NONE && job->hw_decode & HB_DECODE_SUPPORT_FORCE_HW))
+        if (job == NULL || (pv->hw_frame==NULL && job->hw_decode & HB_DECODE_SUPPORT_MF) ||
```

Comment:
Probably it would be better to just set HB_DECODE_SUPPORT_FORCE_HW if the goal is to always enable the hardware decoder.

Reply:
Tried setting `hw_decode` to `HB_DECODE_SUPPORT_FORCE_HW` but it gets reset to 0 by `video_decode_support` during scan.

Also the goal is not to enable hw decoder all the time but to let users choose if they want to use it by passing the `--enable-hw-decoding=mf` option.

Reply:
Did you try with HB_DECODE_SUPPORT_MF |  HB_DECODE_SUPPORT_FORCE_HW ? It's a bitfield to you have to set both.

There is some code that disables the hardware decoders if software only filters are used, so the only way to always use an hardware decoder is to set HB_DECODE_SUPPORT_FORCE_HW too. Unfortunately there isn't a cli option to set it at the moment, but it would be good to add one.

All the others hw decoders work the same way: there is an option to enable them that will enable them only if there is a full hardware accelerated pipeline, and another option to always enable them.

If it's still being disabled somewhere else, please point out where it's being disabled.

I would prefer the addition of an additional cli option to set HB_DECODE_SUPPORT_FORCE_HW, instead of hard-coding it like you did.

Reply:
Yeah, setting it to HB_DECODE_SUPPORT_MF | HB_DECODE_SUPPORT_FORCE_HW worked.